src/components/PrivateRoute.jsx

```diff
@@ -1,19 +1,53 @@
-import React from 'react';
+import '@reshuffle/code-transform/macro';
+import React, { useState, useEffect } from 'react';
 import { Route, Redirect } from 'react-router-dom';
+import { useAuth } from '@reshuffle/react-auth';
+import { validateUser } from '../../backend/authBackend';
 
 /**
- * Standard React-router route that requires a userToken
+ * Standard React-router route that requires  reshuffle identity
  */
-const PrivateRoute = ({ props, component: Component, userToken, ...rest }) => {
+const PrivateRoute = ({ props, component: Component, ...rest }) => {
+  const { authenticated } = useAuth();
+  const [email, setEmail] = useState(false);
+  const url = window.location.href;
+  const urlArr = url.split('/');
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const value = await validateUser();
+      if (value.error) {
+        setEmail(false);
+      } else {
+        setEmail(value);
+      }
+    };
+    fetchData();
+  }, []);
+
+  if (authenticated === undefined) {
+    return null;
+  }
+
   return (
     <Route
       {...rest}
-      render={(routeProps) => {
-        if (userToken) {
-          return (<Component userToken={userToken} {...routeProps} {...props} />);
-        } else {
-          // if userToken is not found, redirect to auth page
-          return (<Redirect to='/auth'/>);
+      render={routeProps => {
+        if (authenticated && email) {
+          return <Component {...routeProps} {...props} />;
+        } else if (authenticated && !email) {
+          // if authenticated is false, redirect to auth page
+          return <Redirect to='/' />;
+        } else if (!authenticated) {
+          // if authenticated is false, redirect to auth page
+          return (
+            <Redirect
+              to={{
+                pathname: '/auth',
+                state: { path: urlArr[3] },
```

Comment:
This is a bit hard to understand for an outsider like myself, who doesn't know the exact structure of paths in this app.  Can we do better than access the fourth element of the path in `url`?  E.g. can we use [`path-to-regexp`](https://www.npmjs.com/package/path-to-regexp) and name path components?  Or, at least, add a comment explaining the structure of the URL and why we to access that specific component.

Reply:
This conversation is _not_ resolved.

Reply:
Not sure why the changes didn't appear here, maybe a human error and forgot to push changes to the branch, but I decided to use the routerv4 `pathname` prop to pass that instead of getting the url and getting the fourth element which was pretty confusing and also added some comments that hopefully make it more clearer on what I'm doing, sorry for closing this conversation again!